### PR TITLE
Realloc instructions

### PR DESCRIPTION
**Pull request by ebatsell**
* also standardizes on epoch over ncn_epoch in instructions

notes to self
* can remove restaking_config from some accounts
* need to update ncn_epoch->epoch variable name in a few places still

### CODE REVIEW

File: program/src/realloc_weight_table.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        let mut weight_table_data = weight_table.try_borrow_mut_data()?;
+        weight_table_data[0] = WeightTable::DISCRIMINATOR;
+        let weight_table_account =
+            WeightTable::try_from_slice_unchecked_mut(&mut weight_table_data)?;
+
+        *weight_table_account =
+            WeightTable::new(*ncn.key, epoch, Clock::get()?.slot, weight_table_bump);
+
+        weight_table_account.initalize_weight_table(&unique_mints)?;
+        msg!("INTIIALIZZING");
```
Comment (coachchucksol) [inline program/src/realloc_weight_table.rs:68]: nit remove

**Review by coachchucksol — COMMENTED** [review]
A couple things to finish up

File: core/src/epoch_snapshot.rs [hunk 2]
```diff
@@ -223,6 +227,7 @@ impl Discriminator for OperatorSnapshot {
 
 impl OperatorSnapshot {
     pub const MAX_VAULT_OPERATOR_STAKE_WEIGHT: usize = 64;
```
[the remark this answers is not in the document]
Comment (ebatsell) [reply core/src/epoch_snapshot.rs]: removed

File: program/src/realloc_operator_snapshot.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        realloc(operator_snapshot, new_size, payer, &Rent::get()?)?;
+    }
+
+    if operator_snapshot.data_len() >= OperatorSnapshot::SIZE
+        && operator_snapshot.try_borrow_data()?[0] != OperatorSnapshot::DISCRIMINATOR
+    {
+        let current_slot = Clock::get()?.slot;
+        let (_, ncn_epoch_length) = load_ncn_epoch(restaking_config, current_slot, None)?;
+
+        //TODO move to helper function
```
[the remark this answers is not in the document]
Comment (coachchucksol) [reply program/src/realloc_operator_snapshot.rs:78]: nit: TODO

File: program/src/route_ncn_rewards.rs [hunk 4]
```diff
@@ -65,8 +63,12 @@ pub fn process_route_ncn_rewards(
 
     let rent_cost = ncn_reward_router_account.rent_cost(&Rent::get()?)?;
 
+    msg!("A");
```
Comment (coachchucksol) [inline program/src/route_ncn_rewards.rs:66]: Nit: Remove

**Review by coachchucksol — APPROVED** [review]
A few nits otherwise lgtm

**Pull request merged**